changelog

tjschuckrug opened the pull request:
Added option to upsample peak in case of Gaussian integration. Relevant for GhOST instrument because of small dwell time values.

tjschuckrug committed:
changelog

Amsoht commented:
How urgent is this? do you want me to test this or simply approve it? Note that the master branch is not protected for this repository so you would be able to merge this without any review.

Amsoht commented:
I am asking, because it would take a couple of days until I would have time to test this.

tjschuckiau commented:
no urgent request

Amsoht commented on Integration/call_integration.pro:
```diff
@@ -4,26 +4,29 @@
 
   IF NOT KEYWORD_SET(plot) THEN plot = 0
   IF NOT KEYWORD_SET(fix_xyrange) THEN fix_xyrange = 0
+  ; IF NOT KEYWORD_SET(gau_upsample) THEN gau_upsample = 0
```
should this be active?

Amsoht commented on Integration/call_integration.pro:
```diff
@@ -71,6 +74,7 @@ PRO call_integration, sel_chrom, sel_name, PLOT=plot, FIX_XYRANGE=fix_xyrange, M
   IF FINITE(chk_noise) NE 1 THEN chk_noise = 0. ; set zero if not calculated yet
 
   verbose = 0
+  upsample =  1
```
should this be removed?

tjschuckrug replied:
Yes, variable is not needed. This was a line remaining from tests.

Amsoht commented on Integration/int_gau.pro:
```diff
@@ -25,6 +27,7 @@ FUNCTION int_gau, xval, yval, NSIGMA_FIT=nsigma_fit, NSIGMA_INT=nsigma_int, NTER
   IF timescale EQ 'Minutes' THEN timescale = 60. ELSE timescale = 1. ; 60: minutes / 1: seconds
   nterms = nterms_base + 3 ; nterms paramenter for gaussfit function
   ; datapoints per second? -> gauss min sigma
+  IF NOT keyword_set(upscale) THEN upscale = 0.
```
do you mean ´upsample´?

tjschuckrug replied:
Actually, yes. Oops. Will be changed.

Amsoht commented on Integration/int_gau.pro:
```diff
@@ -35,6 +38,8 @@ FUNCTION int_gau, xval, yval, NSIGMA_FIT=nsigma_fit, NSIGMA_INT=nsigma_int, NTER
       wdth: !VALUES.D_NAN, $
       ts: !VALUES.D_NAN, $
       te: !VALUES.D_NAN, $
+      upscale: 1 ,$
```
´upscale´ is always 1 this way?

Amsoht replied:
this gets overwritten by ´scale´ later in l207, I see it's only an initialization

tjschuckrug replied:
Yes, this should remain as is, but I will add an explanatory comment

tjschuckrug committed:
Minor corrections according to comments on pull request

Implemented changes according to review of earlier pull request:

-- int_gau.pro: added explanation to initialisation of upscaling factor in structure

-- call_integration.pro: included keyword definition for missing keyword case

-- call_integration.pro: removed initiaisation of unused variable upsample
